### ISSUE TITLE
callback structure unclear

### ISSUE THREAD

**Issue by markvwd** [AuraQ/AutoCompleteForMendix#11]
Debugging the widget JavaScript code I found some callback structures that were unclear to me (and possible incorrect).

Am i correct in assuming that in the _loadCurrentValue function the callback is called twice?

Once here:

```
if(referencedObjectGuid !== null && referencedObjectGuid !== "") {                        
    mx.data.get({
        guid: referencedObjectGuid,
        callback: dojoLang.hitch(this, function(obj){                             
            this._processResults([obj],this._formatCurrentValue, callback);              
        })
    });
} else{                
    if (callback && typeof callback === "function") {                
        callback();
    }
};
```

and once here:

```
// Execute callback.
if (typeof callback !== 'undefined') {
    callback();
}
```

**Comment by lindski** [comment]
The callback at the end of that method does indeed appear to be unnecessary. I'll test this a bit more but it looks like it will get called twice incorrectly when there is a referenced guid present

**Comment by lindski** [comment]
Removed the callback and retested. Fixed in commit fee00ae

Will be added to the next release. In the meantime, you can use the mpk from the dist folder
